gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/grpc.py.j2

```diff
@@ -101,6 +104,27 @@ class {{ service.name }}GrpcTransport({{ service.name }}Transport):
           google.api_core.exceptions.DuplicateCredentialArgs: If both ``credentials``
               and ``credentials_file`` are passed.
         """
+    
+        # If a custom API endpoint is set, set scopes to ensure the auth
+        # library does not used the self-signed JWT flow for service
+        # accounts
+        if (
+            host is not None
+            and self.DEFAULT_HOST is not None
+            and host.split(":")[0] != self.DEFAULT_HOST.split(":")[0]
+            and not scopes
```

Comment:
Why does this logic need to be here and in `base.py`? I also see it two more times below for `grpc_async`. Is that entirely necessary? Can we make this DRYer? Four times for the same logic makes me nervous.

Reply:
That is a fair point. I opened up https://github.com/googleapis/gapic-generator-python/pull/808/ to try to remove some of the existing duplication, which should make it possible to shrink this PR as well.